### PR TITLE
Connectors/salesforce

### PR DESCRIPTION
**Pull request by jonathanlukas**
## Description

There is a new connector incoming: Salesforce Connector!

https://github.com/camunda/connectors/pull/1059

## When should this change go live?

- [x] This change is not yet live and should not be merged until 12 September (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/connectors/out-of-the-box-connectors/salesforce.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+---
+
+The Salesforce Connector is an outbound protocol Connector that allows you to connect your BPMN service with [Salesforce](https://salesforce.com/) to interact with the [Salesforce APIs](https://developer.salesforce.com/docs/apis).
+
+## Prerequisites
+
+To use the **Salesforce Connector**, you must have a [Salesforce Connected App with OAuth 2.0 Client Credentials Flow](https://help.salesforce.com/s/articleView?id=sf.connected_app_client_credentials_setup.htm&type=5).
+
+:::note
+It is highly recommended not to expose your Salesforce Connected App Client Id and Client Secret as plain text. Instead, use Camunda secrets. Learn more in our documentation on [managing secrets](/components/console/manage-clusters/manage-secrets.md).
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/salesforce.md:14]:
```suggestion
It is highly recommended not to expose your Salesforce connected application client ID and client secret as plain text. Instead, use Camunda secrets. Learn more in our documentation on [managing secrets](/components/console/manage-clusters/manage-secrets.md).
```

File: docs/components/connectors/out-of-the-box-connectors/salesforce.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Each operation required information about the **Salesforce Base URL**.
+
+Example: `https://MyDomainName.my.salesforce.com`
+
+The Salesforce API version should be the one you want to use. You can search for this information [in your Salesforce API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_versions.htm).
+
+## Authentication
+
+In the **Authentication** section, select **Bearer Token** to provide a static access token or **OAuth 2** to configure Client Credentials.
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/salesforce.md:33]:
```suggestion
In the **Authentication** section, select **Bearer Token** to provide a static access token or **OAuth 2.0** to configure Client Credentials.
```

File: docs/components/connectors/out-of-the-box-connectors/salesforce.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+:::note
+While the static access token is useful for getting started, it is highly recommended to provide the **OAuth 2** Client Credentials.
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/salesforce.md:36]:
```suggestion
While the static access token is useful for getting started, it is highly recommended to provide the **OAuth 2.0** client credentials.
```

File: docs/components/connectors/out-of-the-box-connectors/salesforce.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
+---
+id: salesforce
+title: Salesforce Connector
+description: Manage your Salesforce Instance from your BPMN process. Learn how to create a Salesforce Connector task, and get started.
+---
+
+The Salesforce Connector is an outbound protocol Connector that allows you to connect your BPMN service with [Salesforce](https://salesforce.com/) to interact with the [Salesforce APIs](https://developer.salesforce.com/docs/apis).
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/salesforce.md:7]:
```suggestion
The **Salesforce Connector** is an outbound protocol Connector that allows you to connect your BPMN service with [Salesforce](https://salesforce.com/) to interact with the [Salesforce APIs](https://developer.salesforce.com/docs/apis).
```

File: docs/components/connectors/out-of-the-box-connectors/salesforce.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+:::note
+While the static access token is useful for getting started, it is highly recommended to provide the **OAuth 2** Client Credentials.
+:::
+
+## Operation
+
+### Types of Operation
+
+Currently, the connector supports 2 types of operation:
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/salesforce.md:43]:
```suggestion
Currently, the Connector supports 2 types of operation:
```

File: docs/components/connectors/out-of-the-box-connectors/salesforce.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+In the **Authentication** section, select **Bearer Token** to provide a static access token or **OAuth 2** to configure Client Credentials.
+
+:::note
+While the static access token is useful for getting started, it is highly recommended to provide the **OAuth 2** Client Credentials.
+:::
+
+## Operation
+
+### Types of Operation
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/salesforce.md:41]:
```suggestion
### Operation types
```

File: versioned_docs/version-8.2/components/connectors/out-of-the-box-connectors/salesforce.md
Comment (igpetrov) [inline versioned_docs/version-8.2/components/connectors/out-of-the-box-connectors/salesforce.md:1]: Same changes as suggested above

**Review by igpetrov — APPROVED** [review]
LGTM from the engineering side. Minor suggestions.

File: docs/components/connectors/out-of-the-box-connectors/salesforce.md (hunk 1 above)
Comment (jonathanlukas) [reply docs/components/connectors/out-of-the-box-connectors/salesforce.md:14]: Salesforce Connected App is the name of a product. I made it _italic_ to show this.

**Review by igpetrov — APPROVED** [review]
LGTM, thank you

**Comment by igpetrov** [comment]
Good to go after https://github.com/camunda/web-modeler/pull/5987 is merged

**Pull request merged**